### PR TITLE
Add reset password function

### PR DESCRIPTION
**Pull request by Nguyen-Ba-Cong**
Please check my code and give me review.
Thank you very much

### CODE REVIEW

File: DEMO-PROJECT-BE/DP/appsettings.json [hunk 1]
```diff
@@ -8,8 +8,9 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "AppCnn": "Data Source=ADMIN; Initial Catalog= souvenir_store ; User ID=ADMIN; Integrated Security=True;"
+    "AppCnn": "Data Source=DESKTOP-1JO6K8V\\SQLEXPRESS; Initial Catalog= souvenir_store ; Integrated Security=True;"
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP/appsettings.json:11]: You can use .\\\SQLEXPRESS or change it locally and don't commit this file.

File: DEMO-PROJECT-BE/DP/Controllers/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            if (account == null)
+            {
+                response.Code = 404;
+                response.IsSuccess = false;
+                response.Message = "Email Not Found";
+                return BadRequest(response);
+            }
+            else
+            {
+                Random rand = new Random();
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP/Controllers/AccountController.cs:78]: put this business code inside a service

File: DEMO-PROJECT-BE/DP.Repository/Implement/AccountRepository.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            }
+            return null;
+        }
+
+        public void SavePassword(int id, string password)
+        {
+            try
+            {
+                var account = GetAccountById(id);
+                account.Password = password;
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/AccountRepository.cs:111]: Did you encrypt  the password?

File: DEMO-PROJECT-BE/DP.Service/Implement/AccountService.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        public bool ChangePassword(int id, ChangePasswordModel changePasswordModel)
+        {
+            if (changePasswordModel.ConfirmPassword != changePasswordModel.NewPassword)
+            {
+                return false;
+            }
+            else
+            {
+                var oldPassword = _accountRepository.GetOldPassword(id);
+                if (changePasswordModel.OldPassword != oldPassword)
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Service/Implement/AccountService.cs:31]: please compare 2 encrypted password

File: DEMO-PROJECT-BE/DP.Repository/Implement/AccountRepository.cs [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            try
+            {
+                var accountChange = GetAccountById(id);
+                if (accountChange == null)
+                {
+                    return false;
+                }
+                else
+                {
+                    accountChange.Password = password;
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/AccountRepository.cs:32]: encrypt the password, check for Jessie code

**Review by vancuong196 — CHANGES_REQUESTED** [review]

File: DEMO-PROJECT-BE/DP/appsettings.json (hunk 1 above)
Comment (Nguyen-Ba-Cong) [reply DEMO-PROJECT-BE/DP/appsettings.json:11]: I added the Sendgrid key in this file.  Do I need to commit this file?

Comment (vancuong196) [reply DEMO-PROJECT-BE/DP/appsettings.json:11]: Yes but remove your device credentials using .\\SQLEXPRESS

File: DEMO-PROJECT-BE/DP.Repository/Implement/RoleRepository.cs [hunk 6]
```diff
@@ -19,23 +17,28 @@ public class RoleRepository : IRoleRepository
...
-            foreach (var role in roles)
+            try
             {
-                if (role.Name != null)
+                List<Role> roles = GetRoles(id);
+                var check = false;
+                foreach (var role in roles)
                 {
-                    if (role.Name.Equals("ADMIN")) check = true;
+                    if (role.Name != null)
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/RoleRepository.cs:29]: Use String.IsNullOrEmpty to check null or empty. 
You can use "Admin".equals(role) to remove this additional null check.
Make role ADMIN as a const string.

File: DEMO-PROJECT-BE/DP.Infrastructure/obj/project.nuget.cache [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Infrastructure/obj/project.nuget.cache:1]: Ingore

File: DEMO-PROJECT-BE/DP.Infrastructure/obj/project.assets.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Infrastructure/obj/project.assets.json:1]: Ignore

**Review by vancuong196 — CHANGES_REQUESTED** [review]

File: DEMO-PROJECT-BE/DP.Repository/Implement/RoleRepository.cs [hunk 9]
```diff
@@ -26,11 +27,12 @@ public bool CheckRole(int id)
                 var check = false;
                 foreach (var role in roles)
                 {
-                    if (role.Name != null)
+                    if (role.Name.Equals(ADMIN))
```
Comment (vancuong196) [inline DEMO-PROJECT-BE/DP.Repository/Implement/RoleRepository.cs:30]: ADMIN.equals()

**Pull request closed without merging**